systemdspawner/systemdspawner.py

```diff
@@ -157,16 +158,14 @@ def __init__(self, *args, **kwargs):
 
         systemd_version = systemd.get_systemd_version()
         if systemd_version is None:
-            self.log.warning(
-                "Failed to parse systemd version from 'systemctl --version'"
-            )
+            warnings.warn("Failed to parse systemd version from 'systemctl --version'")
```

Comment:
If this warning were moved to inside `get_systemd_version`, it could include the error and output that failed to be parsed.

Reply:
Yikes I'm struggling...

systemd.py doesn't have an ability for me to reference `self.log`, and I figured I'd emit `self.log.critical` at one point there if failing to call `systemctl` at all - not just failing to parse the response. What do you think makes sense to do?

```python
def get_systemd_version():
    """
    Returns systemd's major version, or None if failing to do so.
    """
    try:
        version_response = subprocess.check_output(["systemctl", "--version"])
    except subprocess.CalledProcessError:
        warnings.warn("Failed to call 'systemctl --version', is systemd installed?")
        raise

    try:
        # Example response to parse from Ubuntu 22.04:
        #
        # systemd 249 (249.11-0ubuntu3.9)
        # +PAM +AUDIT +SELINUX +APPARMOR +IMA +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBFDISK +PCRE2 -PWQUALITY -P11KIT -QRENCODE +BZIP2 +LZ4 +XZ +ZLIB +ZSTD -XKBCOMMON +UTMP +SYSVINIT default-hierarchy=unified
        #
        version = int(float(version_response.split()[1]))
        return version
    except:
        warnings.warn(f"Failed to parse systemd version from 'systemctl --version' response: {version_response}")
        return None

```

Reply:
@minrk can push a commit to this PR adjusting it as you think makes sense?

Reply:
Pushed basically exactly what you had there.

Reply:
Thank you!!